### PR TITLE
feat/javadoc: automate Javadoc deployment via gh-pages

### PR DESCRIPTION
**Pull request by samirromdhani**
**Update** `automate_javadoc.yml`

**Fix maven plugin :** 

- https://bugs.openjdk.org/browse/JDK-8215582
   The --no-module-directories option is removed.
- https://github.com/eclipse/microprofile-fault-tolerance/issues/512 
     Refactoring all instances of `<h3>` to `<h2>`

**Require :** 
-  #228

### CODE REVIEW

File: .github/workflows/automate_javadoc.yml [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
 #
 # SPDX-License-Identifier: Apache-2.0
 name: Continuous Deployment - JavaDocs
 
 on:
-  push:
-    branches:
-      - 'feat/javadoc'
+  pull_request_target:
+    types: [ labeled, closed ]
```
Comment (samirromdhani) [inline .github/workflows/automate_javadoc.yml:8]: Hi @AliouDIAITE 

I would like you to read this PR.

Whenever a developer creates a pull request from any branch to `develop`, he must have already set the label `javadoc` or `documentation`.

After her PR is merged, `automate_javadoc.yml ` is triggered, which leads to the creation of a new PR from _depondabot_ to the **gh-pages** branch and it's automatically assigned to the author of the merged one.

Then the gh-pages deployment is ready to update javadoc once he accepts the _depondabot_ PR.

I also mentioned that `gh-pages` actually doesn't have a documentation folder (docs), this need #228

File: .github/workflows/automate_javadoc.yml [hunk 2]
```diff
@@ -1,27 +1,29 @@
...
+
 jobs:
   build:
+    if: ${{  github.event.pull_request.base.ref == 'develop' && github.actor != 'dependabot[bot]' && github.event.pull_request.merged && contains(github.event.pull_request.labels.*.name, 'javadoc') }}
     name: Java Docs
     runs-on: ubuntu-latest
     env:
-      from_branch: feat/javadoc
-      target_branch: feat/117_Documentation_SCD_Generation_Process
+      DEPLOYMENT_BRANCH: gh-pages
```
Comment (AliouDIAITE) [inline .github/workflows/automate_javadoc.yml:16]: To be discussed, we could try to ovoid many branches to manage if possible

Comment (samirromdhani) [reply .github/workflows/automate_javadoc.yml:16]: Yes, there is only the `gh-pages` branch here. 

Capture d’écran du 2023-02-02 15-59-35

See (https://github.com/com-pas/compas-sct/issues/159#issuecomment-1406709340) for more detail.

Comment (AliouDIAITE) [reply .github/workflows/automate_javadoc.yml:16]: ok

**Review by AliouDIAITE — APPROVED** [review]

**Pull request merged**